components/script/dom/xmlhttprequest.rs

```diff
@@ -1068,6 +1067,7 @@ impl XMLHttpRequest {
             _ => { return None; }
         }
         temp_doc.set_encoding_name(DOMString::from(charset.name()));
+        self.response_xml.set(Some(temp_doc.r()));
```

Comment:
Unfortunately this still isn't the correct behavior yet; we need to check whether `self.response_xml` exists first before we commit to any sort of action - see steps [here](https://xhr.spec.whatwg.org/#document-response).